### ISSUE TITLE
[JENKINS-40279] Disabling of Jenkins setup wizard not working as expected

### ISSUE THREAD

**Issue by jenkins-infra-bot** [timja-org/jenkins-gh-issues-poc-11-15#6382]
<p>We are deploying and installing Jenkins for customers using a framework of ours, where we are packaging Tomcat, Jenkins, plugins and other maintenance tools to have a full server up and running. </p>

<p>To be able to do this in a controlled manner for the customer, in for example an upgrade situation, we want to disable the setup wizard. </p>

<p><a href="https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties</a><br/>
jenkins.install.runSetupWizard: "<em>Set to false to skip install and upgrade wizards.</em>"</p>

<p>We are setting the system property <b>jenkins.install.runSetupWizard</b> to false to skip both install and upgrade wizards.</p>

<table><tr><td><p>JAVA_OPTS:<br/>
-Xms500m -Xmx500m -XX:MetaspaceSize=128M -XX:MaxMetaspaceSize=1024M  -XX:+HeapDumpOnOutOfMemoryError -XX:+UseCompressedOops -XX:+UseG1GC -XX:MaxGCPauseMillis=400 -XX:GCPauseIntervalMillis=8000 -Djava.awt.headless=true -Dhudson.model.ParametersAction.keepUndefinedParameters=true -Djenkins.install.runSetupWizard=false -javaagent:./dist/ssl/sys-props-agent.jar=props:./dist/ssl/private/syspropsagent.properties -XX:HeapDumpPath=/<path-to-logs>/logs/</p></td></tr></table>

<p>But still the setup wizard is appearing when upgrading like below. </p>

<p>Going from:<br/>
Tomcat 7.0.50, Jenkins 1.651.3<br/>
Going to:<br/>
Tomcat 8.5.5, Jenkins 2.19.1, -Djenkins.install.runSetupWizard=false</p>

<p>Upgrading in our terms means: <br/>
/usr/<old-installation-of-framework-incl-tomcat-7-and-jenkins-1.651.3><br/>
/usr/<new-installation-of-framework-incl-tomcat-8-and-jenkins-2.19.1><br/>
/usr/in_production/<link-to-active-installation><br/>
/proj/jenkins_home/ [ JENKINS_HOME is kept separated from installation; plugins, jobs, customer data etc ]</p>

<p>Stop old Tomcat/Jenkins, install new Tomcat/Jenkins in parallel, <b>change symlinks (old->new installation)</b>, start new Tomcat/Jenkins. <br/>
And yes, we understand we might be doing a (parallel) install in your terms, when we perform the above "upgrade". </p>

<p>1) We simply can't understand under which situations this system property is supposed to DISABLE the setup wizard? Is it not supposed to be working when running Jenkins in Tomcat? It IS described to "skip install <b>and</b> upgrade wizards". </p>

<p>As we somehow needed to get this to work we have done extensive googling and trying out several approaches where users seem to have struggling with this as well. One workaround tested was this: <br/>
<a href="https://github.com/geerlingguy/ansible-role-jenkins/issues/50" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/geerlingguy/ansible-role-jenkins/issues/50</a></p>

<p>Where we ONLY are adding the mentioned file with the value of "2.0" (no newline)</p>

<p>Create the file JENKINS_HOME/jenkins.install.UpgradeWizard.state with one line, 2.0<br/>
( echo -n 2.0 > $JENKINS_HOME/jenkins.install.UpgradeWizard.state )<br/>
VOILA, no setup wizard appearing after login to upgraded Jenkins! </p>

<p>2) Is this a supported, recommended way of disabling the wizard? Are there any pitfalls going this way? Or are we simply working around a bug here? </p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/thomas_w.svg" title="thomas_w's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=thomas_w">thomas_w</a>, imported from: <a href="https://issues.jenkins.io/browse/JENKINS-40279" target="_blank">Disabling of Jenkins setup wizard not working as expected</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Major
<li><b>component(s)</b>: core
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 8
<li><b>watchers</b>: 10
<li><b>imported</b>: 2025-11-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We are deploying and installing Jenkins for customers using a framework of ours, where we are packaging Tomcat, Jenkins, plugins and other maintenance tools to have a full server up and running. </p>

<p>To be able to do this in a controlled manner for the customer, in for example an upgrade situation, we want to disable the setup wizard. </p>

<p><a href="https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties" class="external-link" target="_blank" rel="nofollow noopener">https://wiki.jenkins-ci.org/display/JENKINS/Features+controlled+by+system+properties</a>
jenkins.install.runSetupWizard: "<em>Set to false to skip install and upgrade wizards.</em>"</p>

<p>We are setting the system property <b>jenkins.install.runSetupWizard</b> to false to skip both install and upgrade wizards.</p>

<div class="panel" style="border-width: 1px;"><div class="panelContent">
<p>JAVA_OPTS:
-Xms500m -Xmx500m -XX:MetaspaceSize=128M -XX:MaxMetaspaceSize=1024M  -XX:+HeapDumpOnOutOfMemoryError -XX:+UseCompressedOops -XX:+UseG1GC -XX:MaxGCPauseMillis=400 -XX:GCPauseIntervalMillis=8000 -Djava.awt.headless=true -Dhudson.model.ParametersAction.keepUndefinedParameters=true -Djenkins.install.runSetupWizard=false -javaagent:./dist/ssl/sys-props-agent.jar=props:./dist/ssl/private/syspropsagent.properties -XX:HeapDumpPath=/&lt;path-to-logs&gt;/logs/</p>
</div></div>

<p>But still the setup wizard is appearing when upgrading like below. </p>

<p>Going from:
Tomcat 7.0.50, Jenkins 1.651.3
Going to:
Tomcat 8.5.5, Jenkins 2.19.1, -Djenkins.install.runSetupWizard=false</p>

<p>Upgrading in our terms means: 
/usr/&lt;old-installation-of-framework-incl-tomcat-7-and-jenkins-1.651.3&gt;
/usr/&lt;new-installation-of-framework-incl-tomcat-8-and-jenkins-2.19.1&gt;
/usr/in_production/&lt;link-to-active-installation&gt;
/proj/jenkins_home/ [ JENKINS_HOME is kept separated from installation; plugins, jobs, customer data etc ]</p>

<p>Stop old Tomcat/Jenkins, install new Tomcat/Jenkins in parallel, <b>change symlinks (old-&gt;new installation)</b>, start new Tomcat/Jenkins. 
And yes, we understand we might be doing a (parallel) install in your terms, when we perform the above "upgrade". </p>

<p>1) We simply can't understand under which situations this system property is supposed to DISABLE the setup wizard? Is it not supposed to be working when running Jenkins in Tomcat? It IS described to "skip install <b>and</b> upgrade wizards". </p>

<p>As we somehow needed to get this to work we have done extensive googling and trying out several approaches where users seem to have struggling with this as well. One workaround tested was this: 
<a href="https://github.com/geerlingguy/ansible-role-jenkins/issues/50" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/geerlingguy/ansible-role-jenkins/issues/50</a></p>

<p>Where we ONLY are adding the mentioned file with the value of "2.0" (no newline)</p>

<p>Create the file JENKINS_HOME/jenkins.install.UpgradeWizard.state with one line, 2.0
( echo -n 2.0 &gt; $JENKINS_HOME/jenkins.install.UpgradeWizard.state )
VOILA, no setup wizard appearing after login to upgraded Jenkins! </p>

<p>2) Is this a supported, recommended way of disabling the wizard? Are there any pitfalls going this way? Or are we simply working around a bug here? </p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
- Docker version 1.10.3, build 20f81dd<br/>
- CentOS Linux release 7.2.1511 (Core)<br/>
- Java(TM) SE Runtime Environment, 1.8.0_45-b14<br/>
- Tomcat 8.5.5<br/>
- Jenkins 2.19.1, -Djenkins.install.runSetupWizard=false
```
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-40279?focusedId=279738&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-279738">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Huh. Seems I got this one wrong when documenting it.</p>

<p>Try adding -Djenkins.install.UpgradeWizard.show=false to disable the upgrade wizard. IIUC this is the one you're concerned about.</p>
</pre>
</details>

</details>
<p>Huh. Seems I got this one wrong when documenting it.</p>

<p>Try adding -Djenkins.install.UpgradeWizard.show=false to disable the upgrade wizard. IIUC this is the one you're concerned about.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/thomas_w.svg" title="thomas_w's avatar" /> thomas_w's <a href="https://issues.jenkins.io/browse/JENKINS-40279?focusedId=279774&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-279774">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Hi Daniel, and thanks for your quick attention to our issues!</p>

<p>To be clear about what we are trying to achieve, it is not to only to hide the wizard, but to stop it from executing at all. </p>

<p>We tried adding the system property -Djenkins.install.UpgradeWizard.show=false , both this property alone and in conjunction with  -Djenkins.install.runSetupWizard=false ,unfortunately with no effect. The setup wizard still appears. </p>

<p>We had a quick look at the implementation and it seems like the method reading out the value from jenkins.install.UpgradeWizard.show is never called. Or did we miss something here?</p>

<p>When this didn't work we dug deeper to find a workable solution for our issue. In the below file we found the handling of the runSetupWizard option a bit strange.
<a href="https://github.com/jenkinsci/jenkins/blob/stable-2.19/core/src/main/java/jenkins/install/InstallUtil.java" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/blob/stable-2.19/core/src/main/java/jenkins/install/InstallUtil.java</a>
If creating file jenkins.install.InstallUtil.lastExecVersion either with value "0.0" or "1.0" you will jump into the part of the code where the handling of jenkins.install.runSetupWizard=false is implemented. And allowing us to bypass the execution of the wizard like we wanted. </p>

<p>After bypassing the setup wizard the file jenkins.install.InstallUtil.lastExecVersion is updated with the current Jenkins version. When performing future upgrades of Jenkins the file will be used and versions (current and future) compared in part of the code where it is unable to skip the setup wizard. If we do not reset the value to "0.0" (or "1.0") prior to the upgrade. Is it really the intention?</p>

<p>Is this a better work around of our issue than the previous one mentioned (jenkins.install.UpgradeWizard.state)?</p>

<p>Thanks in advance!</p>
</pre>
</details>

</details>
<p>Hi Daniel, and thanks for your quick attention to our issues!</p>

<p>To be clear about what we are trying to achieve, it is not to only to hide the wizard, but to stop it from executing at all. </p>

<p>We tried adding the system property -Djenkins.install.UpgradeWizard.show=false , both this property alone and in conjunction with  -Djenkins.install.runSetupWizard=false ,unfortunately with no effect. The setup wizard still appears. </p>

<p>We had a quick look at the implementation and it seems like the method reading out the value from jenkins.install.UpgradeWizard.show is never called. Or did we miss something here?</p>

<p>When this didn't work we dug deeper to find a workable solution for our issue. In the below file we found the handling of the runSetupWizard option a bit strange.<br/>
<a href="https://github.com/jenkinsci/jenkins/blob/stable-2.19/core/src/main/java/jenkins/install/InstallUtil.java" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jenkins/blob/stable-2.19/core/src/main/java/jenkins/install/InstallUtil.java</a><br/>
If creating file jenkins.install.InstallUtil.lastExecVersion either with value "0.0" or "1.0" you will jump into the part of the code where the handling of jenkins.install.runSetupWizard=false is implemented. And allowing us to bypass the execution of the wizard like we wanted. </p>

<p>After bypassing the setup wizard the file jenkins.install.InstallUtil.lastExecVersion is updated with the current Jenkins version. When performing future upgrades of Jenkins the file will be used and versions (current and future) compared in part of the code where it is unable to skip the setup wizard. If we do not reset the value to "0.0" (or "1.0") prior to the upgrade. Is it really the intention?</p>

<p>Is this a better work around of our issue than the previous one mentioned (jenkins.install.UpgradeWizard.state)?</p>

<p>Thanks in advance!</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-40279?focusedId=279782&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-279782">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<blockquote><p>unfortunately with no effect. The setup wizard still appears</p></blockquote>

<p>Please confirm that this is about the Setup Wizard (which takes over the Jenkins UI), and <em>not</em> the Upgrade Wizard (which is a banner).</p>

<p>Also, after Jenkins is available, go to the /systemInfo URL and confirm the System properties are correctly passed to Jenkins. Not sure how that works with Tomcat.</p>
</pre>
</details>

</details>
<blockquote><p>unfortunately with no effect. The setup wizard still appears</p></blockquote>

<p>Please confirm that this is about the Setup Wizard (which takes over the Jenkins UI), and <em>not</em> the Upgrade Wizard (which is a banner).</p>

<p>Also, after Jenkins is available, go to the /systemInfo URL and confirm the System properties are correctly passed to Jenkins. Not sure how that works with Tomcat.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/danielbeck.png" title="danielbeck's avatar" /> danielbeck's <a href="https://issues.jenkins.io/browse/JENKINS-40279?focusedId=279790&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-279790">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Seems I misread the source code. It's not a system property but related to the session. Very unusual.</p>

<p>It looks like there's no support way to skip the upgrade wizard entirely. It's not coming up in usual programmatic situations, so apparently the idea was that it's not too difficult to refuse/go through the wizard and remove the plugins afterwards if undesired.</p>

<p>I'll fix the documentation to match reality. Sorry about the confusion.</p>
</pre>
</details>

</details>
<p>Seems I misread the source code. It's not a system property but related to the session. Very unusual.</p>

<p>It looks like there's no support way to skip the upgrade wizard entirely. It's not coming up in usual programmatic situations, so apparently the idea was that it's not too difficult to refuse/go through the wizard and remove the plugins afterwards if undesired.</p>

<p>I'll fix the documentation to match reality. Sorry about the confusion.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/thomas_w.svg" title="thomas_w's avatar" /> thomas_w's <a href="https://issues.jenkins.io/browse/JENKINS-40279?focusedId=279910&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-279910">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>Sorry, but we're not only after documentation fixes, Daniel. We think we have two real bugs right now:</p>

<p>1) Disabling the Setup Wizard doesn't work as expected when using system property -Djenkins.install.runSetupWizard=false</p>

<p>Problematic code part we think should be reviewed.</p>
<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-java">https:<span class="code-comment">//github.com/jenkinsci/jenkins/blob/stable-2.19/core/src/main/java/jenkins/install/InstallUtil.java
</span>...
    <span class="code-comment">// tests need <span class="code-keyword">this</span> to be 1.0
</span>    <span class="code-keyword">private</span> <span class="code-keyword">static</span> <span class="code-keyword">final</span> VersionNumber NEW_INSTALL_VERSION = <span class="code-keyword">new</span> VersionNumber(<span class="code-quote">"1.0"</span>);
    <span class="code-keyword">private</span> <span class="code-keyword">static</span> <span class="code-keyword">final</span> VersionNumber FORCE_NEW_INSTALL_VERSION = <span class="code-keyword">new</span> VersionNumber(<span class="code-quote">"0.0"</span>);
...
        VersionNumber lastRunVersion = <span class="code-keyword">new</span> VersionNumber(getLastExecVersion());

        <span class="code-comment">// Neither the top level config or the lastExecVersionFile have a version
</span>        <span class="code-comment">// stored in them, which means it's a <span class="code-keyword">new</span> install.
</span>        <span class="code-keyword">if</span> (FORCE_NEW_INSTALL_VERSION.equals(lastRunVersion) || lastRunVersion.compareTo(NEW_INSTALL_VERSION) == 0) {      &lt;=== When does <span class="code-keyword">this</span> <span class="code-keyword">case</span> normally happen?
            Jenkins j = Jenkins.getInstance();
            
            <span class="code-comment">// Allow <span class="code-keyword">for</span> skipping			&lt;=== If-clause working as expected when reached
</span>            <span class="code-keyword">if</span>(shouldNotRun) {
                <span class="code-keyword">try</span> {
                    InstallState.INITIAL_SETUP_COMPLETED.initializeState();
                    <span class="code-keyword">return</span> j.getInstallState();
                } <span class="code-keyword">catch</span> (RuntimeException e) {
                    <span class="code-keyword">throw</span> e;
                } <span class="code-keyword">catch</span> (Exception e) {
                    <span class="code-keyword">throw</span> <span class="code-keyword">new</span> RuntimeException(e);
                }
            }
...
        }
</pre>
</div></div>

<p>Similar issues where this issue have been touched upon/mentioned/seen:
<a href="https://issues.jenkins-ci.org/browse/JENKINS-34007" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-34007</a>
<a href="https://issues.jenkins-ci.org/browse/JENKINS-34035" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-34035</a></p>

<p>2) Suppressing the Upgrade Wizard doesn't work using "session variable" -Djenkins.install.UpgradeWizard.show=false
Unfortunately this approach number two given to us didn't help us forward. We set it the ordinary way using JAVA_OPTS, right or wrong? 
As mentioned earlier, we can't find the methods accessing this "session variable" being called. </p>

<p>Do you want us to file a separate ticket on this? </p>
</pre>
</details>

</details>
<p>Sorry, but we're not only after documentation fixes, Daniel. We think we have two real bugs right now:</p>

<p>1) Disabling the Setup Wizard doesn't work as expected when using system property -Djenkins.install.runSetupWizard=false</p>

<p>Problematic code part we think should be reviewed.</p>

<pre>
https:<span class="code-comment">//github.com/jenkinsci/jenkins/blob/stable-2.19/core/src/main/java/jenkins/install/InstallUtil.java
</span>...
    <span class="code-comment">// tests need <span class="code-keyword">this</span> to be 1.0
</span>    <span class="code-keyword">private</span> <span class="code-keyword">static</span> <span class="code-keyword">final</span> VersionNumber NEW_INSTALL_VERSION = <span class="code-keyword">new</span> VersionNumber(<span class="code-quote">"1.0"</span>);
    <span class="code-keyword">private</span> <span class="code-keyword">static</span> <span class="code-keyword">final</span> VersionNumber FORCE_NEW_INSTALL_VERSION = <span class="code-keyword">new</span> VersionNumber(<span class="code-quote">"0.0"</span>);
...
VersionNumber lastRunVersion = <span class="code-keyword">new</span> VersionNumber(getLastExecVersion());

<span class="code-comment">// Neither the top level config or the lastExecVersionFile have a version
</span><span class="code-comment">// stored in them, which means it's a <span class="code-keyword">new</span> install.
</span><span class="code-keyword">if</span> (FORCE_NEW_INSTALL_VERSION.equals(lastRunVersion) || lastRunVersion.compareTo(NEW_INSTALL_VERSION) == 0) {      <=== When does <span class="code-keyword">this</span> <span class="code-keyword">case</span> normally happen?
    Jenkins j = Jenkins.getInstance();
    
    <span class="code-comment">// Allow <span class="code-keyword">for</span> skipping			<=== If-clause working as expected when reached
</span>    <span class="code-keyword">if</span>(shouldNotRun) {
<span class="code-keyword">try</span> {
    InstallState.INITIAL_SETUP_COMPLETED.initializeState();
    <span class="code-keyword">return</span> j.getInstallState();
} <span class="code-keyword">catch</span> (RuntimeException e) {
    <span class="code-keyword">throw</span> e;
} <span class="code-keyword">catch</span> (Exception e) {
    <span class="code-keyword">throw</span> <span class="code-keyword">new</span> RuntimeException(e);
}
    }
...
}
</pre>

<p>Similar issues where this issue have been touched upon/mentioned/seen:<br/>
<a href="https://issues.jenkins-ci.org/browse/JENKINS-34007" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-34007</a><br/>
<a href="https://issues.jenkins-ci.org/browse/JENKINS-34035" class="external-link" target="_blank" rel="nofollow noopener">https://issues.jenkins-ci.org/browse/JENKINS-34035</a></p>

<p>2) Suppressing the Upgrade Wizard doesn't work using "session variable" -Djenkins.install.UpgradeWizard.show=false<br/>
Unfortunately this approach number two given to us didn't help us forward. We set it the ordinary way using JAVA_OPTS, right or wrong? <br/>
As mentioned earlier, we can't find the methods accessing this "session variable" being called. </p>

<p>Do you want us to file a separate ticket on this? </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/ringerc.jpg" title="ringerc's avatar" /> ringerc's <a href="https://issues.jenkins.io/browse/JENKINS-40279?focusedId=368638&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-368638">comment</a>:</i></summary>


<details><summary><sub><i>Raw content of original comment:</i></sub></summary>

<pre>
<p>I've also been utterly stumped by this, trying incantations with derivative Docker images of the official <tt>jenkins/jenkins:lts</tt> image like</p>

<p>{{RUN echo $JENKINS_VERSION | tee \
    /usr/share/jenkins/ref/jenkins.install.UpgradeWizard.state \
    /usr/share/jenkins/ref/jenkins.install.InstallUtil.lastExecVersion}}</p>

<p>and</p>

<p><tt>ENV JAVA_OPTS "<del>Djenkins.install.runSetupWizard=false ${JAVA_OPTS:</del>}"</tt></p>

<p>But for some reason passing `--env JAVA_OPTS="-Djenkins.install.runSetupWizard=false"` to `docker run` did work. </p>

<p>See also <a href="https://github.com/jenkinsci/docker/issues/310" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/docker/issues/310</a> and <a href="https://github.com/jenkinsci/docker/pull/833" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/docker/pull/833</a></p>
</pre>
</details>

</details>
<p>I've also been utterly stumped by this, trying incantations with derivative Docker images of the official <tt>jenkins/jenkins:lts</tt> image like</p>

<p>{{RUN echo $JENKINS_VERSION | tee \<br/>
    /usr/share/jenkins/ref/jenkins.install.UpgradeWizard.state \<br/>
    /usr/share/jenkins/ref/jenkins.install.InstallUtil.lastExecVersion}}</p>

<p>and</p>

<p><tt>ENV JAVA_OPTS "<del>Djenkins.install.runSetupWizard=false ${JAVA_OPTS:</del>}"</tt></p>

<p>But for some reason passing `--env JAVA_OPTS="-Djenkins.install.runSetupWizard=false"` to `docker run` did work. </p>

<p>See also <a href="https://github.com/jenkinsci/docker/issues/310" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/docker/issues/310</a> and <a href="https://github.com/jenkinsci/docker/pull/833" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/docker/pull/833</a></p>
